x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -115,22 +96,41 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) {
 		}
 
 		svcCloudwatch := cloudwatch.New(*m.MetricSet.AwsConfig)
+		namespace := "AWS/EC2"
+		listMetricsInput := &cloudwatch.ListMetricsInput{Namespace: &namespace}
+		reqListMetrics := svcCloudwatch.ListMetricsRequest(listMetricsInput)
+
+		// List metrics of EC2 for each region
+		listMetricsOutput, err := reqListMetrics.Send()
+		if err != nil {
+			err = errors.Wrap(err, "ListMetricsRequest failed, skipping region "+regionName)
+			m.logger.Error(err.Error())
+			report.Error(err)
+			continue
+		}
+
+		if listMetricsOutput.Metrics == nil || len(listMetricsOutput.Metrics) == 0 {
+			// No EC2 buckets in this region
+			continue
+		}
```

Comment:
Could this code be reused between metricsets? this seems to depend in the namespace only.